partiql-ast/src/experimental/ast.rs

```diff
@@ -112,25 +212,140 @@ pub enum ExprKind {
     Intersect(Intersect),
     /// Other expression types
     Path(Path),
-    Call(Call),
-    CallAgg(CallAgg),
-    Cast(Cast),
-    CanCast(CanCast),
-    CanLossLessCast(CanLossLessCast),
-    NullIf(NullIf),
-    Coalesce(Coalesce),
 
     /// `SELECT` and its parts.
     Select(Select),
 }
 
 #[derive(Clone, Debug, PartialEq)]
 pub struct Lit {
-    pub value: OwnedValue,
+    pub kind: LitKind,
+}
+
+/// `LitKind` is mostly inspired by SQL-92 Literals standard.
+/// See section 5.3 in the following:
+/// https://www.contrib.andrew.cmu.edu/~shadow/sql/sql1992.txt
+#[derive(Clone, Debug, PartialEq)]
+pub enum LitKind {
+    NumericLit(NumericLit),
+    CharStringLit(CharStringLit),
+    NationalCharStringLit(NationalCharStringLit),
+    BitStringLit(BitStringLit),
+    HexStringLit(HexStringLit),
+    DateTimeLit(DateTimeLit),
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct NumericLit {
+    pub kind: NumericLitKind,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum NumericLitKind {
+    Int16(Int16),
+    Int32(Int32),
+    Int64(Int64),
+    Decimal(Decimal),
+    Numeric(Decimal),
+    Real(Real),
+    Float(Float),
+    Double(Double),
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Int8 {
+    pub value: i8,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Int16 {
+    pub value: i16,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Int32 {
+    pub value: i32,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Int64 {
+    pub value: i64,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Decimal {
+    pub value: RustDecimal,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Numeric {
+    pub value: RustDecimal,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Real {
+    pub value: f32,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Float {
+    pub value: f32,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Double {
+    pub value: f64,
 }
 
 #[derive(Clone, Debug, PartialEq)]
-pub struct Id {
+pub struct CharStringLit {
+    value: String,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct NationalCharStringLit {
+    pub value: String,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct BitStringLit {
+    pub value: String,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct HexStringLit {
+    pub value: String,
+}
```

Comment:
Since these `Lit` fields have only a single value, they are effectively 'newtype's. 
Should we consider using [tuple structs](https://doc.rust-lang.org/book/ch05-01-defining-structs.html#using-tuple-structs-without-named-fields-to-create-different-types) for them? Or even inlining them directly in their containing enum?

Reply:
Was thinking perhaps we need to enrich them later down the track with additional data (code span, etc.). What do you think?